src/admin/resolvers/mutations/ApprovedItem.ts

```diff
@@ -245,3 +259,123 @@ export async function uploadApprovedItemImage(
   const image = await data;
   return await uploadImageToS3(context.s3, image);
 }
+
+/**
+ * Imports an approved item
+ * Creates an approved item if it doesn't exist, and
+ * creates a scheduled item for the approved item
+ * if it doesn't exist
+ * @param parent
+ * @param data
+ * @param context
+ */
+export async function importApprovedItem(
+  parent,
+  { data },
+  context: IContext
+): Promise<ImportApprovedCuratedCorpusItemPayload> {
+  // Check if user is authorized to import an item
+  if (!context.authenticatedUser.canWriteToCorpus()) {
+    throw new AuthenticationError(ACCESS_DENIED_ERROR);
+  }
+
+  // Check the language. Throw an error if not valid
+  checkLanguage(data.language);
+
+  // Get approved item
+  // Try creating the approved item first. The assumption here is that for an
+  // import, we don't expect the approved item to exist. Handle an existing
+  // approved item in the catch statement
+  let approvedItem: ApprovedItem;
+  try {
+    approvedItem = await dbImportApprovedItem(
+      context.db,
+      toDbApprovedItemInput(data)
+    );
+    context.emitReviewedCorpusItemEvent(
+      ReviewedCorpusItemEventType.ADD_ITEM,
+      approvedItem
+    );
+  } catch (e) {
+    approvedItem = (await getApprovedItemByUrl(
+      context.db,
+      data.url
+    )) as ApprovedItem;
+  }
+
+  // Get scheduled item
+  // Try creating the scheduled item first. The assumption here is that for an
+  // import, we don't expect the scheduled item to exist. Handle an existing
+  // scheduled item in the catch statement
+  let scheduledItem: ScheduledItem;
+  try {
+    scheduledItem = await importScheduledItem(
+      context.db,
+      toDbScheduledItemInput({
+        ...data,
+        approvedItemId: approvedItem.id,
+      })
+    );
+    context.emitScheduledCorpusItemEvent(
+      ScheduledCorpusItemEventType.ADD_SCHEDULE,
+      scheduledItem
+    );
+  } catch (e) {
+    scheduledItem = (await getScheduledItemByUniqueAttributes(context.db, {
+      approvedItemId: approvedItem.id,
+      scheduledSurfaceGuid: data.scheduledSurfaceGuid,
+      scheduledDate: data.scheduledDate,
+    })) as ScheduledItem;
+  }
+
+  return {
+    approvedItem,
+    scheduledItem,
+  };
+}
+
+/**
+ * Transform GraphQL ImportApprovedCuratedCorpusItemInput to ImportApprovedItemInput
+ * for the database
+ * @param data
+ */
+function toDbApprovedItemInput(
+  data: ImportApprovedCuratedCorpusItemInput
+): ImportApprovedItemInput {
+  return {
+    title: data.title,
+    excerpt: data.excerpt,
+    status: data.status,
+    language: data.language,
+    publisher: data.publisher,
+    imageUrl: data.imageUrl,
+    topic: data.topic,
+    url: data.url,
+    isCollection: data.isCollection,
+    isSyndicated: data.isSyndicated,
+    source: data.source,
+    createdAt: fromUnixTime(data.createdAt),
+    createdBy: data.createdBy,
+    updatedAt: fromUnixTime(data.updatedAt),
+    updatedBy: data.updatedBy,
+  };
```

Comment:
nit: if we wanted to make this short I think we could've done this but this is fine too
`return { ...data, createdAt: ..., updatedAt: ...}`